pkg/util/credentials.go

```diff
@@ -58,6 +100,19 @@ func GetAdminCredentials(secrets map[string]string) (*Credentials, error) {
 	return getCredentials(credAdminID, credAdminKey, secrets)
 }
 
+func GetCredentials(id, key string) (*Credentials, error) {
```

Comment:
Nit: personally don't like very similar named methods doing different things (for clarity purposes). Perhaps rename ```getCredentials``` internal method to something like ```getCredentialsFromSecrets```

Reply:
Ack! will address in the next PR.

Along the same lines should we rename `GetXXXCredentials` to `NewXXXCredentials` so that `DeleteXXXCredentials` is its destructor?

Reply:
> Along the same lines should we rename `GetXXXCredentials` to `NewXXXCredentials` so that `DeleteXXXCredentials` is its destructor?

Makes sense to me since it helps to imply it needs to be cleaned up after use via the `DeleteXXXCredentials`

Reply:
Modified as per provided comments.